examples/region_of_interest/ExtractLineProfile.py

```diff
@@ -0,0 +1,63 @@
+"""
+Extract line profile from image interactively
+=============================================
+
+Interactively extract a line profile (with a certain width) from an image using
+:py:class:`~.api.roi.Line2DROI`. Use :func:`~.api.plot.plot_spectra` to plot several
+line profiles on the same figure. Save a profile data as ``msa`` file.
+
+"""
+
+#%%
+#.. figure::  ../../_static/interactive_profiles.gif
+#    :align:   center
+#    :width:   1024
+#    :alt:     Interactive example gif of region of interest and associated profile plots.
+#
+#    Extracting line profiles and interactive plotting.
+
+#%%
+# Initialize image data as HyperSpy signal:
+import hyperspy.api as hs
+import holospy as holo
+im0 = holo.data.Fe_needle_reference_hologram()
+im1 = holo.data.Fe_needle_hologram()
+
+#%%
+# Intialize Line-ROI from position (400,250) to position (220,600) of width 5
+# in calibrated axes units (in the current example equal to the image pixels):
+line_roi = hs.roi.Line2DROI(400, 250, 220, 600, 5)
+
+#%%
+# Plot the image and display the ROI (creates a new signal object). You can also
+# display the same ROI on a second image to make sure that a profile is well
+# placed on both images:
+im0.plot()
+roi1D = line_roi.interactive(im0, color='green')
+im1.plot()
+roi1D = line_roi.interactive(im1, color='green')
+
+#%%
+# You can then drag and drop the ends of the ROI to adjust the placement.
+#
+# If you want to later update the ROI initialization with the modified parameters
+# you can print these:
+print(tuple(line_roi))
+  
+#%%
+# Extract data along the ROI as new signal by "slicing" the signal and plot the
+# profile:
+profile = line_roi(im0)
+profile.plot()
+
+#%%
+# Extract data along the same ROI from the second image and plot both profiles in
+# a single plot:
+profile2 = line_roi(im1)
+hs.plot.plot_spectra([profile, profile2])
+# Choose the fourth figure as gallery thumbnail:
+# sphinx_gallery_thumbnail_number = 4
+
+#%%
+# Save profile as `.msa` text file:
+profile.save('extracted-line-profile.msa', format='XY')
```

Comment:
This will generate a file when building the documentation. I don't know if sphinx-gallery has a way to deal with this (from a quick search on internet, I couldn't find anything), if not, simply add it to `.gitignore`?

Reply:
Yes, already done for .msa files in another subfolder of examples. (Being at it I reordered `.gitignore` for an easier overview)